content/knowledge-base/guides/write-a-composition-function-in-python.md

```diff
@@ -0,0 +1,749 @@
+* [Hatch](https://hatch.pypa.io/), a Python build tool. This guide uses v1.7.
+* [Docker Engine](https://docs.docker.com/engine/). This guide uses Engine v24.
+* The [Crossplane CLI](https://docs.crossplane.io/latest/cli) v1.14 or newer. This guide uses Crossplane
+  CLI v1.14.
+
+You don't need access to a Kubernetes cluster or a Crossplane control plane to
+build or test a composition function.
+
+## Initialize the function from a template
+
+Use the `crossplane beta xpkg init` command to initialize a new function. When
+you run this command it initializes your function using
+[this GitHub repository](https://github.com/crossplane/function-template-python)
+as a template.
+
+```shell {copy-lines=1}
+crossplane beta xpkg init function-xbuckets https://github.com/crossplane/function-template-python -d function-xbuckets
+Initialized package "function-xbuckets" in directory "/home/negz/control/negz/function-xbuckets" from https://github.com/crossplane/function-template-python/tree/bfed6923ab4c8e7adeed70f41138645fc7d38111 (main)
+```
+
+The `crossplane beta init xpkg` command creates a directory named
+`function-xbuckets`. When you run the command the new directory should look like
+this:
+
+```shell {copy-lines=1}
+ls function-xbuckets
+Dockerfile  example/  function/  LICENSE  package/  pyproject.toml  README.md  renovate.json  tests/
+```
+
+Your function's code lives in the `function` directory:
+
+```shell {copy-lines=1}
+ls function/
+__version__.py  fn.py  main.py
+```
+
+The `function/fn.py` file is where you add the function's code. It's useful to
+know about some other files in the template:
+
+* `function/main.py` runs the function. You don't need to edit `main.py`.
+* `Dockerfile` builds the function runtime. You don't need to edit `Dockerfile`.
+* The `package` directory contains metadata used to build the function package.
+
+{{<hint "tip">}}
+<!-- vale gitlab.FutureTense = NO -->
+<!--
+This tip talks about future plans for Crossplane.
+-->
+In v1.14 of the Crossplane CLI `crossplane beta xpkg init` just clones a
+template GitHub repository. In a future release the command will automate tasks
+like replacing the template name with the new function's name. See Crossplane
+issue [#4941](https://github.com/crossplane/crossplane/issues/4941) for details.
+<!-- vale gitlab.FutureTense = YES -->
+{{</hint>}}
+
+Edit `package/crossplane.yaml` to change the package's name before you start
+adding code. Name your package `function-xbuckets`.
+
+Some functions accept a configuration input. You configure the input when you
+write a Composition that uses the function. The `package/input` directory
+defines the OpenAPI schema for the a function's input. 
+
+The function in this guide doesn't accept an input. For this function you
+should delete the `package/input` directory. The
+[composition functions](https://docs.crossplane.io/latest/concepts/composition-functions)
+documentation explains more the input to a composition function.
```

Comment:
```suggestion
The `package/input` directory
defines the OpenAPI schema for the a function's input. 

The function in this guide doesn't accept an input. Delete the `package/input` 
directory.   
The
[composition functions](https://docs.crossplane.io/latest/concepts/composition-functions)
documentation explains more the input to a composition function.
```

I'd simplify the instructions and let the note below provide the context about inputs.